test/hotspot/jtreg/vmTestbase/nsk/share/jvmti/hotswap/HotSwap.cpp

```diff
@@ -141,9 +141,20 @@ CompiledMethodLoad(jvmtiEnv *jvmti_env, jmethodID method,
         const jvmtiAddrLocationMap* map, const void* compile_info) {
     char *name = nullptr;
     char *signature = nullptr;
+    jvmtiPhase phase;
 
     CompiledMethodLoadEventsCount++;
 
+    // GetMethodNamme is work only in live phasem so just exit
+    // if event is generated too late
+    if (!NSK_JVMTI_VERIFY(jvmti_env->GetPhase(&phase))) {
+        nsk_jvmti_setFailStatus();
+        return;
+    }
+    if (phase == JVMTI_PHASE_DEAD) {
+      NSK_DISPLAY0("CompiledMethodLoad event recieved in dead phase");
+      return;
+    }
```

Comment:
Nit: Typos at L148: `GetMethodNamme` => `GetMethodName`, `is work` => `works`, `phasem` => `phase`
A suggestion is to reformulate the comment as below:
```
    // GetMethodName works in live phase only so just exit if the event is generated too late
```

Also, I wonder if we want to abort/fail in all cases when `phase == JVMTI_PHASE_DEAD`.

Reply:
> Also, I wonder if we want to abort/fail in all cases when phase == JVMTI_PHASE_DEAD.
I take it back. My suggestion does not look right.

Reply:
comment fixed.
I haven't find in spec that event can't be generated in dead phase. So test should just ignore this event.